### PR TITLE
esm-user-login-demo: using TypeScript and ES6 

### PR DESCRIPTION
**Pull request by esride-nik**

### CODE REVIEW

File: esm-user-login-demo/index.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+    showRouteDirections(directions);
+
+    view.ui.empty("top-right");
+    view.ui.add(directionsElement, "top-right");
+  }
+
+  route
+    .solve(routeUrl, routeParams)
+    // reponse of type __esri.DirectionsLastRoute
```
Comment (jf990) [inline esm-user-login-demo/index.ts]:
```suggestion
    // response of type __esri.DirectionsLastRoute
```

File: esm-user-login-demo/index.js [hunk 2]
```diff
@@ -12,100 +12,100 @@ import RouteParameters from "@arcgis/core/rest/support/RouteParameters";
...
         view.ui.empty("top-right");
         view.ui.add(directionsElement, "top-right");
     }
-
-    route.solve(routeUrl, routeParams)
-      .then((response) => {
-        showRoutes(response.routeResults)
+    route
+        .solve(routeUrl, routeParams)
+        // reponse of type __esri.DirectionsLastRoute
```
Comment (jf990) [inline esm-user-login-demo/index.js:87]:
```suggestion
        // response of type __esri.DirectionsLastRoute
```

**Review by jf990 — CHANGES_REQUESTED** [review]
Can you also include an update to the README to change http://localhost:3000 to https://locahost:3000 the oauth callback didn't work for me until I changed that in my app registration

File: esm-user-login-demo/index.js (hunk 2 above)
Comment (esride-nik) [reply esm-user-login-demo/index.js:87]: Sure! Sorry I didn't see your comments before 🙈 need to work on my collaboration skills.

**Review by jf990 — APPROVED** [review]
i ran this branch locally and everything worked as expected. thanks!

**Pull request merged**